src/Queue.js

```diff
@@ -3,88 +3,113 @@ const { SchedulerInterrupt } = require('./constants/index');
 // A class representation of a process queue that may hold either
 // blocking or non-blocking processes
 class Queue {
-    constructor(scheduler, quantum, priorityLevel, queueType) {
-        this.processes = [];
-        // The queue's priority level; the lower the number, the higher the priority
-        this.priorityLevel = priorityLevel;
-        // The queue's parent scheduler
-        this.scheduler = scheduler;
-        // The queue's allotted time slice; each process in this queue is executed for this amount of time
-        this.quantum = quantum;
-        // A counter to keep track of how much time the queue has been executing so far
-        this.quantumClock = 0;
-        this.queueType = queueType;
-    }
+  constructor(scheduler, quantum, priorityLevel, queueType) {
+    this.processes = [];
+    // The queue's priority level; the lower the number, the higher the priority
+    this.priorityLevel = priorityLevel;
+    // The queue's parent scheduler
+    this.scheduler = scheduler;
+    // The queue's allotted time slice; each process in this queue is executed for this amount of time
+    this.quantum = quantum;
+    // A counter to keep track of how much time the queue has been executing so far
+    this.quantumClock = 0;
+    this.queueType = queueType;
+  }
 
-    // Adds the input process to the queue's list of processes
-    // Also sets the input process's parent queue to this queue
-    // Return the newly added process
-    enqueue(process) {
-        
-    }
+  // Adds the input process to the queue's list of processes
+  // Also sets the input process's parent queue to this queue
+  // Return the newly added process
+  enqueue(process) {
+    this.processes.push(process);
+    process.setParentQueue(this);
+    return process;
```

Comment:
This is 100% fine but another way to write this  in a more "terse" way would be
`process.setParentQueue(this);
  return this.process.push(process)`
This is a small change and not something you have to do just wanted to bring it up for your knowledge